doc/programming/language/examples/_optional-check.spicy

```diff
@@ -1,9 +1,11 @@
 # Automatically generated; edit in Sphinx source code, not here.
-global x: optional<uint64>;  # Unset.
-global b1: bool = x;         # False.
-global b2 = cast<bool>(x);   # False.
+type S = struct {
+    a: int32;
+    b: int32;
+};
 
-if ( x )
-    print "'x' was set";     # Never runs.
-if ( ! x )
-    print "'x' was unset";   # Always runs.
\ No newline at end of file
+global s1: type = typeinfo(bool);
+global s2: type = typeinfo(S);
+global s3: type = S;
```

Comment:
Not a huge fan of  the implicit conversion in `global s3: type = S;` (without any explicit `typeinfo`); could we remove support for this, or is it unavoidable to support `typeinfo` taking either a value or a type?